docker/Dockerfile

```diff
@@ -0,0 +1,116 @@
+FROM --platform=linux/amd64 node:16.13.0 AS frontend
+
+WORKDIR /app/oppia
+
+RUN apt-get update -y && apt-get upgrade -y \
+    curl \
+    git \
+    npm
+
+# Installing packages from the package.json file
+COPY package.json .
+COPY scripts/linters/custom_eslint_checks ./scripts/linters/custom_eslint_checks
+
+RUN npm install -g yarn@1.22.15 --force
+# mathjs ESOCKETTIMEDOUT error - there are cases reported for mathjs package when timeout happens while
+# downloading the package using `yarn install` command.
+# The solution is to increase the network timeout, so that the package is downloaded successfully.
+RUN yarn install --network-timeout 1000000
+
+
+# Using digestID of the 3rd party dockerhub image, as updates in the dockerhub images might break the build
+# Added platform flag here because -- https://stackoverflow.com/questions/71040681/qemu-x86-64-could-not-open-lib64-ld-linux-x86-64-so-2-no-such-file-or-direc
+# python:3.8.17
+FROM --platform=linux/amd64 python:3.8.17-slim-buster AS backend
```

Comment:
yes, we need to specify the platform flag here. please take a look to the above comment with a link for reference

Reply:
Err.. please read my above comment fully! I think the attached links fixes OS-specific issues and considering we are expecting to run the dev env in different OS, are we sure that a specific strict platform won't create issues for different OS/arch? (Similar to what we have seen with sha256 images)

Reply:
correct, I am not sure my reasoning would be correct, but --
check the [dockerhub-image](https://hub.docker.com/layers/library/python/3.8.17-slim-buster/images/sha256-90834dba6381dfc3957573dc7a3e6c5c8ed255cf60079329a6da2b5e6d4257b8?context=explore) we are using.
you can see the OS/arch listed in the dockerhub image are Linux only.. so we are just specifying one of the linux arch.. I was facing this problem and this soln works (working on my mac, ubuntu of my friend, Praneeth's laptop, and I think yours too, so I think this can be done). also note, when we were using the SHA of the dockerhub images, this image was not problematic

Reply:
refer to my answer [here](https://github.com/oppia/oppia/pull/18578/files#r1258984677)